infra/terraform/modules/data_access/memberships/tests/test_detach_bucket_policies.py

```diff
@@ -0,0 +1,58 @@
+import json
+from mock import call
+
+import botocore.exceptions
+import pytest
+
+import memberships
+
+from tests.conftest import (
+    TEST_READ_ONLY_POLICY_ARN,
+    TEST_READ_WRITE_POLICY_ARN,
+    TEST_ROLE_NAME,
+    TEST_TEAM_SLUG,
+    TEST_USERNAME,
+)
+
+
+def raise_detach_role_policy_exception(**kwarg):
+    raise botocore.exceptions.ClientError(
+        {
+            "Error": {
+                "Code": "NoSuchEntity",
+                "Message": "Policy arn:... was not found.",
+            }
+        },
+        "DetachRolePolicy"
+    )
+
+
+@pytest.mark.usefixtures(
+    "given_the_env_is_set",
+    "given_iam_is_available",
+)
+def test_when_invoked_detaches_all_bucket_policies_from_the_role(iam_client_mock):
+    # This is to test the scenario when 1+ of the policies tried to be
+    # detached is not actually attached: In this case we still boto/AWS will
```

Comment:
I need to re-read this comment again as I clearly got distracted while writing it and the sentence doesn't make any sense :P